test/functional/wallet_listsinceblock.py

```diff
@@ -167,6 +168,28 @@ def test_reorg(self):
         found = next(tx for tx in transactions if tx['txid'] == senttx)
         assert_equal(found['blockheight'], self.nodes[0].getblockheader(nodes2_first_blockhash)['height'])
 
+    def test_cant_read_block(self):
+        self.log.info('Test the RPC error "Can\'t read block from disk"')
+
+        # Split network into two
+        self.split_network()
+
+        # generate on both sides
+        nodes1_last_blockhash = self.generate(self.nodes[1], 6, sync_fun=lambda: self.sync_all(self.nodes[:2]))[-1]
+        self.generate(self.nodes[2], 7, sync_fun=lambda: self.sync_all(self.nodes[2:]))[0]
+
+        self.join_network()
+
+        # Renaming the block file to induce unsuccessful block read
+        (self.nodes[0].blocks_path / "blk00000.dat").rename(self.nodes[0].blocks_path / "blk00000.dat.copy")
+
+        # listsinceblock(nodes1_last_blockhash) should now fail as blocks are not accessible
+        assert_raises_rpc_error(-32603, "Can't read block from disk",
+            self.nodes[0].listsinceblock, nodes1_last_blockhash)
+
+        # Restoring block file
+        (self.nodes[0].blocks_path / "blk00000.dat.copy").rename(self.nodes[0].blocks_path / "blk00000.dat")
```

Comment:
re. this comment https://github.com/bitcoin/bitcoin/pull/30195#discussion_r1630237777 and needing to use the full path again, I think it is that `rename(...)` returns a new `Path` instance, which is then the 'correct' one to use going forwards for further changes you want to happen on-disk (alas Python has no 'use of moved value' concept like Rust et al. that would've complained at you trying to do this).

I think the way to think of this is that a `Path` is just a *path*, not a file-handle, so you can have a Path instance that doesn't exist, because you might want to create it -- so `.rename(...)` on a Path that doesn't exist changes the *path instance* but of-course can't change the actual filesystem because it doesn't exist.

```python
>>> from pathlib import Path
>>> # Make a Path and rename it to add `".copy"` on the end:
>>> p = Path("README.md")
>>> p.exists()
True
>>> q = p.rename("README.md.copy")  # capture the return value in `q`
>>>
>>> # We still have the Path instance called `p`...
>>> p
PosixPath('README.md')
>>> # ...but it doesn't exist anymore (becuase we just renamed it)
>>> p.exists()
False
>>> # `.rename` returned a new Path instance, which *does* exist
>>> q
PosixPath('README.md.copy')
>>> q.exists()
True
>>>
>>> # Just to prove the rename happened: 
>>> [item.name for item in Path(".").iterdir() if "md" in item.name]
['README.md.copy']
>>>
>>> # Rename again to move it back to the original name...
>>> r = q.rename("README.md")
>>> r
PosixPath('README.md')
>>> r.exists()
True
>>> # Because `p` and `r` now have the same name, the original `p` also 'exists' again:
>>> p.exists(), q.exists(), r.exists()
(True, False, True)
```

*So:*

```suggestion
        # Renaming the block file to induce unsuccessful block read
        blk_dat = (self.nodes[0].blocks_path / "blk00000.dat")
        assert blk_dat.exists()
        blk_dat_moved = blk_dat.rename("blk00000.dat.moved")
        assert not blk_dat.exists()

        # listsinceblock(nodes1_last_blockhash) should now fail as blocks are not accessible
        assert_raises_rpc_error(-32603, "Can't read block from disk",
            self.nodes[0].listsinceblock, nodes1_last_blockhash)

        # Restoring block file
        blk_dat_moved.rename("blk00000.dat")
        assert blk_dat.exists()
```

Also suggest naming change to "moved" or "backup" instead of "copy" as you're not copying?

And additional `assert`s to check that the files exist and don't when expected (maybe overkill, but prevents unexpected pass/fails for some other reason than you renaming the file)

Reply:
(Could even `blk_dat_moved.rename(blk_dat.name)` to avoid repeating the literal `"blk00000.dat"` but meh)

Reply:
The `exists()` asserts are a nice improvement/sanity check.

Reply:
I was getting this error so I opted to use the full path

error message: `OSError: [Errno 18] Invalid cross-device link: '/tmp/bitcoin_func_test_lt7jywkj/node0/regtest/blocks/blk00000.dat.moved' -> 'blk00000.dat'`

but I like the assertions thanks for the suggestion!
updated in [9bf646c](https://github.com/bitcoin/bitcoin/pull/30195/commits/9bf646c4190e55a13bf3a89d1705763c7a833a06)

Reply:
Ah, d'oh. Full path is fine then, sorry I should've checked with a `Path` not in my current working dir